### ISSUE TITLE
facts(...) with multiple records facts does not output correct failure messages

### ISSUE THREAD

**Issue by SamRoberts** [CommBank/thermometer#45]
I have a `facts` call in a thermometer spec with multiple `records` facts.

When one of those `records` facts failed, the message I get is the concatenation of all the failure messages for each `record` fact, even though only one of those facts failed.

I haven't debugged this properly, but a quick skim of specs suggests that the `updateMessage` call in `PathFact.checkEquality` is a bit misguided. Our call to `updateMessage` seems to assume that the message is a failure message, whereas the specs code for `updateMessage` and `ResultMonoid` appears to suggest that the message applies regardless of whether it is a success or failure.

**Comment by stephanh** [comment]
@SamRoberts any chance of a PR to fix this :smile:?

**Comment by SamRoberts** [comment]
Yeah, definitely, but not in the short term. It's not really a problem for us at the moment. Just something we discovered one day.
